be/src/vec/olap/olap_data_convertor.cpp

```diff
@@ -700,4 +621,55 @@ Status OlapBlockDataConvertor::OlapColumnDataConvertorDecimal::convert_to_olap()
     return Status::OK();
 }
 
-} // namespace doris::vectorized
\ No newline at end of file
+Status OlapBlockDataConvertor::OlapColumnDataConvertorArray::convert_to_olap() {
+    const ColumnArray* column_array = nullptr;
+    const DataTypeArray* data_type_ptr_array = nullptr;
+    if (_nullmap) {
+        const auto* nullable_column =
+                assert_cast<const ColumnNullable*>(_typed_column.column.get());
+        column_array =
+                assert_cast<const ColumnArray*>(nullable_column->get_nested_column_ptr().get());
+        data_type_ptr_array = assert_cast<const DataTypeArray*>(
+                (assert_cast<const DataTypeNullable*>(_typed_column.type.get())->get_nested_type())
+                        .get());
+    } else {
+        column_array = assert_cast<const ColumnArray*>(_typed_column.column.get());
+        data_type_ptr_array = assert_cast<const DataTypeArray*>(_typed_column.type.get());
+    }
+    assert(column_array);
+    assert(data_type_ptr_array);
+
+    const auto& offsets = column_array->get_offsets();
+    CollectionValue* collection_value = _values.data();
+    for (int i = 0; i < _num_rows; ++i, ++collection_value) {
+        int64_t cur_pos = _row_pos + i;
+        int64_t prev_pos = cur_pos - 1;
+        if (_nullmap && _nullmap[cur_pos]) {
+            continue;
+        }
+        auto offset = offsets[prev_pos];
+        auto size = offsets[cur_pos] - offsets[prev_pos];
+        new (collection_value) CollectionValue(size);
+
+        if (size == 0) {
+            continue;
+        }
+
+        if (column_array->get_data().is_nullable()) {
+            const auto& data_nullable_column =
+                    assert_cast<const ColumnNullable&>(column_array->get_data());
+            const auto* data_null_map = data_nullable_column.get_null_map_data().data();
+            collection_value->set_has_null(true);
+            collection_value->set_null_signs(
+                    const_cast<bool*>(reinterpret_cast<const bool*>(data_null_map + offset)));
+        }
+        ColumnWithTypeAndName item_typed_column = {column_array->get_data_ptr(),
+                                                   data_type_ptr_array->get_nested_type(), ""};
+        _item_convertor->set_source_column(item_typed_column, offset, size);
```

Comment:
if item_type is or varchar or string, `set_source_column` will `resize()` the `_item_convertor._slice` and the slices from the previous `convert_to_olap()` will be freed, and then lead to heap use after free in the subsequent use of the collection_value

Reply:
the same as array, hll or bitmap

Reply:
Hi @eldenmoon , you are right.